src/main/proto/netflix/titus/titus_job_api.proto

```diff
@@ -540,6 +540,10 @@ message JobChangeNotification {
     /// Emitted when a task is created or its state has changed.
     message TaskUpdate {
         Task task = 1;
+        /// movedFromAnotherJob will be true on the first event for the target Job after a task is moved between jobs.
+        //  task.jobId will be the destination job, and it will include a 'task.movedFromJob' entry in its taskContext
+        //  map with the source jobId.
+        bool movedFromAnotherJob = 2;
```

Comment:
For `observerJob(<job_id>)`, will this event be emitted for the moved from job? We have to signal it somehow, but this event might be confusing.
For the target job, we have already `task.movedFromJob` attribute, so there is no need for the flag.

Reply:
`observeJob(<jobId>)` will get a single `JobUpdateEvent` with a new (reduced) desired size in the source job. The destination job will get both the `JobUpdateEvent` with the increased desired size, and a `TaskUpdateEvent` with the `movedFromAnotherJob` flag set. I think that is good enough to cover all cases, no?

The problem in relying only on the `task.movedFromJob` taskContext attribute is that it is always present. Consumers of the stream wouldn't have an easy way to detect exactly when the task was moved, and would need to keep local state for the stream, for each task. The main usage of this is for subscribers of the stream that index tasks by their jobId. The flag in the event is an easy way for these subscribers to know when the index needs to be updated.

Reply:
My may concern was about the source job. I have missed the fact that we will have job-level event (job size decreased by one).
I think this proposal is reasonable.